cwms-data-api/src/main/java/cwms/cda/api/TimeSeriesController.java

```diff
@@ -200,7 +163,8 @@ public void create(@NotNull Context ctx) {
             DSLContext dsl = getDslContext(ctx);
 
             TimeSeriesDao dao = getTimeSeriesDao(dsl);
-            TimeSeries timeSeries = deserializeTimeSeries(ctx);
+            TimeSeriesWithDataEntryDate timeSeries = deserializeTimeSeries(ctx);
```

Comment:
This should fail deserialization if the time series doesn't include the data entry date since you are typing to the subclass.

Reply:
I thought so too, but it deserializes with null values for the data entry dates. This seems to be the most reliable way to check for those values, since using the body content as a String or InputStream to check for them would consume the context and close the data stream. I found that the default behavior when passing in a TS with data entry values was to drop/ignore the values and treat it as a standard TS, so typing to the subclass doesn't appear to cause issues since it is cast to the base class anyway for storage. 

I'm open to suggestions for a better way to approach this.